core/document/src/document.rs

```diff
@@ -354,6 +354,14 @@ impl Document {
 				let path = path.as_slice()[..path.len() - 1].to_vec();
 				Some(vec![DocumentResponse::DocumentChanged, DocumentResponse::FolderChanged { path }])
 			}
+			Operation::FillLayer { path, color } => {
+				if let Ok(layer) = self.layer_mut(path) {
+					layer.style.set_fill(Some(layers::style::Fill::new(*color)));
+					layer.cache_dirty = true;
+				}
+				self.root.cache_dirty = true;
```

Comment:
The root doesn't need to be set as dirty or have a `DocumentChanged` operation if the layer does not exist

Reply:
You're right, I have it this way because making both changes in the same scope requires taking `self` as mut twice. I'll rewrite it slightly to avoid that. By the way, this operation is unconditional, I was using `if let` because it separated the scopes in which self was borrowed.

Reply:
I reordered the operations and replaced the `if let` with an `unwrap`. While most operations that receive paths `unwrap` them, `ToggleVisibility` uses Result::map and discards the result (which I think is an antipattern). I imagine the operations should be consistent about this, and it may be a good idea to move away from `unwrap` before it becomes a problem for concurrency or something.